lib/test_prof/before_all.rb

```diff
@@ -16,26 +16,28 @@ class << self
       attr_accessor :adapter
 
       def begin_transaction
+        raise "begin_transaction does not support a block" if block_given?
         raise AdapterMissing if adapter.nil?
 
         config.run_hooks(:begin) do
           adapter.begin_transaction
         end
-        yield
-      end
-
-      def within_transaction
-        yield
       end
 
       def rollback_transaction
-        raise AdapterMissing if adapter.nil?
-
         config.run_hooks(:rollback) do
           adapter.rollback_transaction
         end
       end
 
+      def begin_example_transaction
```

Comment:
Those two methods are used to open/rollback a transaction around examples. The existing ones could have been reused, but this way it seems to be more clear, and no conditional is needed to decide whether to run hooks or not.